### PR TITLE
Move osu! specific settings into its own ConfigManager

### PR DESCRIPTION
**Pull request by Aergwyn**
- [x] depends on #4136 

~~Moved the osu! specific settings into a new `OsuConfigManager` specific for that ruleset. For that I renamed the previous `OsuConfigManager` to `GameConfigManager` (same for the respective setting enums).~~

Renamed ruleset specific classes according to feedback to not have two `OsuConfigManager`s.

### CODE REVIEW

File: osu.Game/Configuration/GameConfigManager.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            Set(GameSetting.ShowStoryboard, true);
+            Set(GameSetting.BeatmapSkins, true);
+            Set(GameSetting.BeatmapHitsounds, true);
+
+            Set(GameSetting.CursorRotation, true);
+
+            Set(GameSetting.MenuParallax, true);
+
+            Set(GameSetting.SnakingInSliders, true);
+            Set(GameSetting.SnakingOutSliders, true);
```
Comment (smoogipoo) [inline osu.Game/Configuration/GameConfigManager.cs]: Wanna remove these now?

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Configuration/GameConfigManager.cs (hunk 1 above)
Comment (Aergwyn) [reply osu.Game/Configuration/GameConfigManager.cs]: :eyes: I screwed up the merge. Yes, will remove the zombies again.

**Comment by peppy** [comment]
I'm thinking we may want to keep the `OsuSetting` enum as-is, to avoid confusion that it may be a framework-level type (because `Game` is a framework level class for the game).

Maybe we should rename the ruleset ones instead to `OsuRulesetSetting` `TaikoRulesetSetting`?

**Comment by Aergwyn** [comment]
Seems fair to me. I'll adjust that later.

**Comment by Aergwyn** [comment]
All feedback and requested changes so far were addressed.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**